### ISSUE TITLE
ErrorDisconnected is not fired when headphones are unplugged the first time

### ISSUE THREAD

**Issue by dturner** [google/oboe#41]
From Lukasz @ ROLI: 

Test environment: Google Pixel running Android API 26, the latest Oboe master (7283080bae61db63c9990aa7682503d76723da03) and Android Studio 3.0.1 with NDK version 16.1.4479499

The app in some circumstances does not receive oboe::Result::ErrorDisconnected when unplugging the headphones for the first time. The consecutive plugging an unplugging triggers the error callback correctly. To reproduce:
- the app should open stream targeting wired headphones at startup (rather than letting the builder to pick a default device)
=> the sound will correctly play in headphones
- unplug headphones
=> there will be no sound, since headphones unplugging gets not picked up
- plug headphones
=> there will be sound in headphones
- unplug headphones
=> now there will be sound in speakers

One way JUCE users can open a device is to simply ask for a default device and then open it. 	The way I’ve done it in JUCE is I open a dummy stream with no device ID, to let Oboe pick the default device, then I check the device ID, close the dummy stream and use that device as a default one. 

To reproduce the same behaviour in Hello Oboe:
- add the lines in bold below to mimic JUCE behaviour:

```
void PlayAudioEngine::createPlaybackStream() {

   oboe::AudioStreamBuilder builder2;
   oboe::AudioStream* testStream = nullptr;
   builder2.openStream (&testStream);
   mPlaybackDeviceId = testStream->getDeviceId();
   testStream->close;

   oboe::AudioStreamBuilder builder;
```

- now unplug the headphones and notice that the change will not be picked up, only after consecutive plugging/unplugging the sound will play correctly

Perhaps the issue could be avoided if there was a function like int `AudioStreamBuilder::getDefaultDeviceId()` which could pick a default device at given situation?

**Comment by philburk** [comment]
This is probably an AAudio bug. Does it happen on O MR1   AKA   V8.1    AKA   API 27 ?
Does it happen in hello-aaudio?

**Comment by lukaszkozakiewicz-roli** [comment]
The issue seems to be fixed on 8.1, however 8.1 seems to have introduced new crashes, as mentioned here: https://github.com/google/oboe/issues/45
